src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -342,6 +342,10 @@ public async Task GetSourceAsync(
 
             string fetchFilter = executionContext.GetInput(Pipelines.PipelineConstants.CheckoutTaskInputs.FetchFilter);
 
+            string sparseCheckoutDirectories = executionContext.GetInput(Pipelines.PipelineConstants.CheckoutTaskInputs.SparseCheckoutDirectories);
+            string sparseCheckoutPatterns = executionContext.GetInput(Pipelines.PipelineConstants.CheckoutTaskInputs.SparseCheckoutPatterns);
+            bool enableSparseCheckout = !string.IsNullOrWhiteSpace(sparseCheckoutDirectories) || !string.IsNullOrWhiteSpace(sparseCheckoutPatterns);
```

Comment:
This should check for some type of "true" boolean value e.g. 1 or 'true' as a change or configuration may use '0' or 'false' (which should not be interpreted as 'true')

Reply:
I'm capturing patterns here so checking for a boolean value would require a third property, which I was trying to avoid. Looking at the checkout task, we have other string properties that enables functionality based on whether there's a value or not.

https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines